### PR TITLE
Store namespace in predicate as a hex separated by a hyphen to prevent json marshal issues

### PR DESCRIPTION
**Pull request by harshil-goel**
We used to store predicate as <namespace>|<attribute> (pipe | signifies concatenation). We store this as a string. <namespace> is 8 bytes uint64, which when marshaled to JSON bytes mess up the predicate. This is because for the namespace greater than 127, the UTF-8 encoding might take up several bytes (also if the mapping does not exist, then it replaces it with some other rune). This affects three identified places in Dgraph:

- Live loader using guardian of galaxy
- Backup and List Backup
- Http clients and Ratel
- Schema and predicate

Fix:
Fix is to have a UTF-8 string when dealing with JSON. A better idea is to use UTF-8 string even for internal operations. Only when we read/write to badger we convert it into the format of the byte.
New Format: <namespace>-<attribute> (- is the hyphen literal) <namespace> is a string "81" in hex

We also update the manifest version after update. This diff takes care that older backups are still compatible and can be used to restore.

### CODE REVIEW

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: worker/backup_ee.go [hunk 1]
```diff
@@ -554,13 +554,13 @@ func (pr *BackupProcessor) CompleteBackup(ctx context.Context, m *Manifest) erro
...
-	manifest, err := GetManifest(handler, uri)
+	manifest, err := GetManifestNoUpgrade(handler, uri)
 	if err != nil {
 		return err
 	}
 	manifest.Manifests = append(manifest.Manifests, m)
 
-	if err := createManifest(handler, uri, manifest); err != nil {
+	if err := CreateManifest(handler, uri, manifest); err != nil {
 		return errors.Wrap(err, "Complete backup failed")
```
Comment (matthewmcneely) [inline worker/backup_ee.go:564]: Realize this is not your doing, but error text should not be capitalized

File: worker/restore_map.go [hunk 2]
```diff
@@ -721,15 +721,14 @@ func RunMapper(req *pb.RestoreRequest, mapDir string) (*mapResult, error) {
...
-						return nil, errors.Wrap(err, "Map phase failed to parse namespace")
-					}
+				if op.DropValue == "" {
+					// In 2103, we do not support namespace level drop data.
+					dropAll = true
+					continue
+				}
+				ns, err := strconv.ParseUint(op.DropValue, 0, 64)
+				if err != nil {
+					return nil, errors.Wrap(err, "Map phase failed to parse namespace")
```
Comment (matthewmcneely) [inline worker/restore_map.go:731]: Error text should not be capitalized

**Review by matthewmcneely — CHANGES_REQUESTED** [review]

File: ee/updatemanifest/run.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
+// +build !oss
+
+/*
+ * Copyright 2021 Dgraph Labs, Inc. and Contributors
```
Comment (mangalaman93) [inline ee/updatemanifest/run.go:4]: fix the license

File: x/keys.go [hunk 4]
```diff
@@ -57,8 +57,23 @@ const (
...
 	NamespaceOffset = 1
+	// NsSeparator is the separator between between the namespace and attribute.
+	NsSeparator = "-"
 )
 
+// Invalid bytes are replaced with the Unicode replacement rune.
+// See https://golang.org/pkg/encoding/json/#Marshal
+const replacementRune = rune('\ufffd')
+
+func AttrFrom2103(attr string) (string, error) {
```
Comment (mangalaman93) [inline x/keys.go:68]: is this function defined again here?

Comment (mangalaman93) [reply x/keys.go:68]: And can we explain what is going on here? maybe an example in the comments.

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

**Review by mangalaman93 — DISMISSED** [review]

**Review by mangalaman93 — APPROVED** [review]

**Review by matthewmcneely — APPROVED** [review]

**Pull request merged**